Added globaloptions parameter to dhcp class

dgmorales opened the pull request:
This allows one to add arbitrary global options to the main dhcpd.conf file, the same way the module already deals with arbitrary options in a specific pool.

tphoney commented:
@dgmorales ping? have you had a chance to look at the tests ?

dgmorales commented:
@tphoney Hello, sorry for my silence. I'll take a look at it in a few days.

I thought I  would be able to do it way sooner, so I didn't answer before. I don't have experience with spec tests yet, but this is a nice opportunity to start changing that.

tphoney commented:
@dgmorales Thanks for the effort you are putting in, there are loads of examples in other modules and http://ask.puppetlabs.com/questions/ is a great resource too.

dgmorales commented:
I've updated this PR with the tests. I've also rebased it to the current master so it applies cleanly. Hope that's OK.

Good learning. In the process I've discovered that dhcp_conf_header was being set to 'eth0', making my tests fail (actually any test based on the content of dhcpd.conf would fail).

I've also tried to refactor spec/classes/dhcp_spec.rb so to set a default param of interface => 'eth0' once in the top, instead of setting it in each context. But that got me to more strange errors that I couldn't understand yet.

mhaskel commented on manifests/init.pp:
```diff
@@ -19,6 +19,7 @@
   $default_lease_time  = 3600,
   $max_lease_time      = 86400,
   $service_ensure      = running,
+  $globaloptions       = ''
```
Could you keep the trailing comma here?

mhaskel commented:
@dgmorales thanks, I had one more comment, and otherwise if you could squash this down to a single commit it should be ready to go!

dgmorales commented:
@mhaskel done, take a look.

mhaskel commented:
Looks good, thanks @dgmorales !
